File: Client/src/app/model/cours.ts
import {Categorie} from './categorie';
import {User} from './user';

export class Cours {
  idCours: number
  titre: string
  description: string
  idCategorie: number
  categorie:Categorie
  dateDebut: Date
  dateFin: Date
  idUser: number
  urlImage: string
  enseignant: User
  nbrInscri : number

  constructor(idCours: number, titre: string, description: string, idCategorie: number, categorie: Categorie, dateDebut: Date, dateFin: Date, idUser: number, urlImage: string, enseignant: User,  nbrInscri : number) {
    this.idCours = idCours;
    this.titre = titre;
    this.description = description;
    this.idCategorie = idCategorie;
    this.categorie = categorie;
    this.dateDebut = dateDebut;
    this.dateFin = dateFin;
    this.idUser = idUser;
    this.urlImage = urlImage;
    this.enseignant = enseignant;
    this.nbrInscri=nbrInscri
  }
}

File: InscriptionCoursServices/src/main/java/ma/ac/emi/coursemi/entities/Cours.java
package ma.ac.emi.coursemi.entities;

import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;


@Data
@Getter@Setter@AllArgsConstructor
public class Cours {

	private Long idCours ;
	
	private String titre;
	
	private String description;
	private Long idCategorie;
	private Categorie categorie;
	private LocalDate dateDebut;
	private LocalDate dateFin;
	private Long idUser;
	private String urlImage;
	private Enseignant enseignant;
	private Integer nbrInscri;
}
